### PR TITLE
+doc document ByteBufferSerializer for Artery #21601

### PR DESCRIPTION
**Pull request by ktoso**
refs  #21601

### CODE REVIEW

File: akka-docs/rst/scala/code/docs/actor/ByteBufferSerializerDocSpec.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import java.nio.charset.StandardCharsets
+
+import akka.serialization.{ByteBufferSerializer, SerializerWithStringManifest}
+
+/**
+ * Requirements are as follows:
+ * The first thing the actor needs to do, is to subscribe to a channel of events,
+ * Then it must replay (process) all "old" events
+ * Then it has to wait for a GoAhead signal to begin processing the new events
+ * It mustn't "miss" events that happen between catching up with the old events and getting the GoAhead signal
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/actor/ByteBufferSerializerDocSpec.scala]: copy pasta

File: akka-docs/rst/scala/code/docs/actor/ByteBufferSerializerDocSpec.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class ByteBufferSerializerDocSpec {
+ 
+  //#bytebuf-with-manifest
+  class ExampleByteBufSerializer extends SerializerWithStringManifest with ByteBufferSerializer {
+    override def identifier: Int = 1337
+    override def manifest(o: AnyRef): String = "naive-toStringImpl"
+
+    // Implement this method for compatibility with `SerializerWithStringManifest`.
+    override def toBinary(o: AnyRef): Array[Byte] = {
+      val buf = ByteBuffer.allocateDirect(256) // in production code, aquire this from a BufferPool
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/actor/ByteBufferSerializerDocSpec.scala]: use `allocate`

**Review by patriknw — APPROVED** [review]
LGTM after nitpick
Then we need a java example also, thanks.

**Comment by ktoso** [comment]
I can do the java example, but not creating yet the remoting-artery.rst right?

**Comment by ktoso** [comment]
Added java example now, but no remoting-artery.rst

**Comment by patriknw** [comment]
I ran akka-docs/test locally

**Pull request merged**